### ISSUE TITLE
Allow for "hiding" intermediate steps from `earthly ls`

### ISSUE THREAD

**Issue by bbugh** [earthly/earthly#2102]
Hello planet earthly 👋🏻 

We have a complex monorepo with nx which is a big pain in the 🍑 and requires a bunch of steps, so our Earthfile has only 4 useful commands and 15 overall commands. Our `earthly ls` shows a lot of different steps that nobody but someone working on the Earthfile would need to know. 

I've been prefixing externally useless intermediate steps with `x-` but it would be nice if there was a way (like prefixing with `_` or some kind of meta setting) that would hide those commands. Imagining:

```
» earthly ls
+all
+build-app1
+build-app2
+lint
+test

(10 steps hidden, use `--all` to view them
```

And with `--all`

```
» earthly ls --all
+all
+build-app1
+build-app2
+lint
+test
+_container-base
+_dependencies
+_hack1
+_hack2
+_image-build-base
+_nx-base
+_nx-build-app
+_nx-build-base
+_nx-build-lib
+_playwright-base
```

**Comment by vladaionescu** [comment]
Feels that this is somewhat related to https://github.com/earthly/earthly/issues/583

**Comment by mwopfner** [comment]
Yes, they are definitely related. I added a comment in #583 which explains how they could be combined.
